src/main/scopes/js/import-matchers/js-all-import-matcher.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright IBM Corp. 2024, 2024
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import type { JsFunction, JsImport, JsImportMatcher, JsToken } from '../interfaces.js'
+
+/**
+ * Identifies JsTokens or JsFunctions that have been imported as all imports,
+ * and returns an import element match (if any) or undefined otherwise.
+ */
+export class JsAllImportMatcher implements JsImportMatcher<JsToken | JsFunction> {
```

Comment:
For these classes that basically just implement an interface, you can keep these class-level docs pretty basic, like "Import matcher for all (*) imports, such as `import * as stuff from 'whatever'`" since it probably makes more sense to put the bulk of the explanation on the method below instead (like you have).